src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -1294,18 +1217,241 @@ class Peer(
+                            val inboundLiquidityTarget = when (val policy = nodeParams.liquidityPolicy.first()) {
+                                is LiquidityPolicy.Disable -> 1.sat
+                                is LiquidityPolicy.Auto -> policy.inboundLiquidityTarget ?: 1.sat
+                            }
+                            // We assume that the liquidity policy is correctly configured to match a funding lease offered by our peer.
+                            val fundingRate = walletParams.remoteFundingRates.findRate(inboundLiquidityTarget)!!
+                            LiquidityAds.RequestFunding(inboundLiquidityTarget, fundingRate, LiquidityAds.PaymentDetails.FromChannelBalance)
+                        }
+                        val (localFundingAmount, fees) = run {
+                            // We need to know the local channel funding amount to be able use channel opening messages.
+                            // We must pay on-chain fees for our inputs/outputs of the transaction: we compute them first
+                            // and proceed backwards to retrieve the funding amount.
+                            val dummyFundingScript = Script.write(Scripts.multiSig2of2(Transactions.PlaceHolderPubKey, Transactions.PlaceHolderPubKey)).byteVector()
+                            val localMiningFee = Transactions.weight2fee(currentFeerates.fundingFeerate, FundingContributions.computeWeightPaid(isInitiator = true, null, dummyFundingScript, cmd.walletInputs, emptyList()))
+                            val localFundingAmount = cmd.totalAmount - localMiningFee
+                            val fundingFees = requestRemoteFunding.fees(currentFeerates.fundingFeerate, isChannelCreation = true)
+                            // We also refund the liquidity provider for some of the on-chain fees they will pay for their inputs/outputs of the transaction.
+                            // This will be taken from our channel balance during the interactive-tx construction, they shouldn't be deducted from our funding amount.
+                            val totalFees = ChannelManagementFees(miningFee = localMiningFee + fundingFees.miningFee, serviceFee = fundingFees.serviceFee)
+                            Pair(localFundingAmount, totalFees)
+                        }
+                        if (cmd.totalAmount - fees.total < nodeParams.dustLimit) {
+                            logger.warning { "cannot create channel, not enough funds to pay fees (fees=${fees.total}, available=${cmd.totalAmount})" }
+                            swapInCommands.trySend(SwapInCommand.UnlockWalletInputs(cmd.walletInputs.map { it.outPoint }.toSet()))
+                        } else {
+                            when (val rejected = nodeParams.liquidityPolicy.first().maybeReject(requestRemoteFunding.requestedAmount.toMilliSatoshi(), fees.total.toMilliSatoshi(), LiquidityEvents.Source.OnChainWallet, logger)) {
```

Comment:
We must include the swapped amount in the liquidity policy check, like we do in the splice case.

```suggestion
                            when (val rejected = nodeParams.liquidityPolicy.first().maybeReject(cmd.walletInputs.balance.toMilliSatoshi() + requestRemoteFunding.requestedAmount.toMilliSatoshi(), fees.total.toMilliSatoshi(), LiquidityEvents.Source.OnChainWallet, logger)) {
```